### PR TITLE
issue: 1541163 Fix VMA_EXCEPTION_HANDLING Error mode

### PR DESCRIPTION
**Pull request by rafiw**
This commit rewrite the logic of VMA_EXCEPTION_HANDLING and move
the handling of this logic to a central place, also fix RETURN_ERROR mode.

Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by rafiw** [comment]
bot:retest

**Comment by rafiw** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

File: src/vma/sock/sockinfo_tcp.cpp [hunk 1]
```diff
@@ -3788,20 +3789,11 @@ int sockinfo_tcp::getsockopt(int __level, int __optname, void *__optval,
...
-		VLOG_PRINTF_INFO(safe_mce_sys().exception_handling.get_log_severity(), "%s", buf);
-		int rc = handle_exception_flow();
-		switch (rc) {
-		case -1:
-			return rc;
-		case -2:
-			vma_throw_object_with_msg(vma_unsupported_api, buf);
+	if (ret != SOCKOPT_HANDLE_BY_OS) {
+		ret = handle_exception_flow(__level, __optname, *__optlen,
+				"setsockopt");
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: getsockopt? Consider passing '__FUNCTION__' or '__func__' (see https://stackoverflow.com/a/18920987)

File: src/vma/util/sys_vars.h [hunk 2]
```diff
@@ -199,6 +199,18 @@ namespace vma_spec {
...
+	VMA_EXCEPTION_MODE_EXIT = -2,
+	VMA_EXCEPTION_MODE_DEBUG = -1,
+	VMA_EXCEPTION_MODE_UNOFFLOAD = 0,
+	VMA_EXCEPTION_MODE_LOG_ERROR_AND_UNOFFLOAD = 1,
+	VMA_EXCEPTION_MODE_RETURN_ERROR = 2,
+	VMA_EXCEPTION_MODE_ABORT = 3,
+	VMA_EXCEPTION_MODE_LAST = 4,
+
+	VMA_EXCEPTION_MODE_DEFAULT = VMA_EXCEPTION_MODE_DEBUG
+} vma_exception_mode;
```
Comment (igor-ivanov) [inline src/vma/util/sys_vars.h]: It looks as coding style change.

**Pull request closed without merging**